packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -215,7 +224,7 @@ export default class InternalModel {
 
   get _recordArrays() {
     if (this.__recordArrays === null) {
-      this.__recordArrays = new OrderedSet();
+      this.__recordArrays = new Set();
```

Comment:
I strongly recommend (I know it's not new here) explicitly naming the return type of this getter.